log4j-1.2-api/src/main/java/org/apache/log4j/Category.java

```diff
@@ -309,15 +309,10 @@ private void fireRemoveAppenderEvent(final Appender appender) {
 
     public void forcedLog(final String fqcn, final Priority level, final Object message, final Throwable t) {
         final org.apache.logging.log4j.Level lvl = org.apache.logging.log4j.Level.toLevel(level.toString());
+        final Message msg = LegacyMessageFactory.INSTANCE.newMessage(message);
         if (logger instanceof ExtendedLogger) {
-            @SuppressWarnings("unchecked")
-            final Message msg = message instanceof Message ? (Message) message
-                : message instanceof Map ? new MapMessage((Map) message) : new ObjectMessage(message);
             ((ExtendedLogger) logger).logMessage(fqcn, lvl, null, msg, t);
         } else {
-            final ObjectRenderer renderer = get(message.getClass());
-            final Message msg = message instanceof Message ? (Message) message
-                : renderer != null ? new RenderedMessage(renderer, message) : new ObjectMessage(message);
```

Comment:
The `RenderedMessage` case is omitted, since this would require a stateful `MessageFactory` (per category).

Anyway there is currently no way to add elements to the `RendererMap`.